api/src/main/java/com/datastrato/gravitino/tag/Tag.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.tag;
+
+import com.datastrato.gravitino.Auditable;
+import com.datastrato.gravitino.annotation.Evolving;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * The interface of a tag. A tag is a label that can be attached to a catalog, schema, table,
+ * fileset, topic, or column. It can be used to categorize, classify, or annotate these entities.
+ */
+@Evolving
+public interface Tag extends Auditable {
+
+  /**
+   * A reserved property to specify the color of the tag. The color is a string of hex code that
+   * represents the color of the tag. The color is used to visually distinguish the tag from other
+   * tags.
+   */
+  String PROPERTY_COLOR = "color";
```

Comment:
For `PROPERTY_COLOR`, should the backend reserve `category` or `class` property?

I think `color` should be determined by the frontend based on the category(or class) of tag. For example, a certain color may be black in normal mode on the frontend, but will change to white in night mode.

Reply:
I think color should be stored in the backend, otherwise there's no way to display the selected color in the frontend.

Frontend can decide how to display based on the selected color and the current mode.